src/main/java/com/playonlinux/ui/api/InstallerFilter.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (C) 2015 Markus Ebner
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+package com.playonlinux.ui.api;
+
+import com.playonlinux.common.dtos.ScriptDTO;
+import com.playonlinux.common.dtos.ScriptInformationsDTO;
+import com.playonlinux.domain.PlayOnLinuxError;
+import com.playonlinux.utils.OperatingSystem;
+import org.apache.commons.lang.StringUtils;
+
+import java.util.Observable;
+
+/**
+ * This class is handling the filtering of Installers, providing one central place for
+ * changing how the filtering is done.
+ * InstallerFilters extends Observable, so that changes get directly promoted and
+ * initiate a new filtering process.
+ */
+public class InstallerFilter extends Observable {
+
+    private String title = null;
+    private String category = null;
+    private boolean showTesting = false;
+    private boolean showNoCd = false;
+    private boolean showCommercial = true;
+
+    private final OperatingSystem hostOs;
+
+
+    public InstallerFilter() {
+        //TODO: Pre-define filters.
+        OperatingSystem os;
+        try {
+            os = OperatingSystem.fetchCurrentOperationSystem();
+        } catch (PlayOnLinuxError err) {
+            os = null;
```

Comment:
I don't really like doing this. Couldn't we manage the exception in a different way? Maybe adding a default operating system policy? At least we need to add a todo here